server/controllers/apply/applicationsController.ts

```diff
@@ -59,11 +60,20 @@ export default class ApplicationsController {
 
       if (eligibilityIsConfirmed(application)) {
         if (consentIsConfirmed(application)) {
-          const { errors, errorSummary } = fetchErrorsAndUserInput(req)
-
-          const referrer = req.headers.referer
-          const taskList = new TaskListService(application)
-          return res.render('applications/taskList', { application, taskList, errors, errorSummary, referrer })
+          if (hdcDatesHaveBeenEntered(application)) {
```

Comment:
feel like the logic around what page to display is starting to get unreadable but after tooling around for a bit I'm not sure how to make it better quickly 😅 
was thinking it could be at least bundled into one function like `showMissingRequiredTasksOrTaskList`, but not sure if that makes it more readable?

Reply:
looks good - at least we've brushed it under the 'utils' rug :)